extensions/amp-accordion/0.1/amp-accordion.js

```diff
@@ -221,40 +228,156 @@ class AmpAccordion extends AMP.BaseElement {
-        // expanded at a time
+        this.animateExpand_(section).then(() => {
+        });
         if (this.element.hasAttribute('expand-single-section')) {
           this.sections_.forEach(sectionIter => {
             if (sectionIter != section) {
-              sectionIter.removeAttribute('expanded');
+              this.animateCollapse_(sectionIter);
               sectionIter.children[0].setAttribute('aria-expanded', 'false');
             }
           });
         }
       } else {
-        section.removeAttribute('expanded');
         header.setAttribute('aria-expanded', 'false');
+        this.animateCollapse_(section);
       }
-    }, section);
+    } else { // Toggle without animation
+      this.mutateElement(() => {
+        if (toExpand) {
+          section.setAttribute('expanded', '');
+          header.setAttribute('aria-expanded', 'true');
+          // if expand-single-section is set, only allow one <section> to be
+          // expanded at a time
+          if (this.element.hasAttribute('expand-single-section')) {
+            this.sections_.forEach(sectionIter => {
+              if (sectionIter != section) {
+                sectionIter.removeAttribute('expanded');
+                sectionIter.children[0].setAttribute('aria-expanded', 'false');
+              }
+            });
+          }
+        } else {
+          section.removeAttribute('expanded');
+          header.setAttribute('aria-expanded', 'false');
+        }
+      }, section);
+    }
     this.currentState_[contentId] = !isSectionClosedAfterClick;
     this.setSessionState_();
   }
 
+  /**
+   * @param {!Element} section
+   * @return {!Promise}
+   * @private
+   */
+  animateExpand_(section) {
+    let height, duration;
+    const sectionChild = section.children[1];
+
+    return this.mutateElement(() => {
+      section.setAttribute('expanded', '');
+      st.setStyles(sectionChild, {
+        opacity: 0,
+      });
+    }).then(() => {
+      return this.measureMutateElement(
+          () => {
+            height = sectionChild./*OK*/offsetHeight;
+            const viewportHeight = this.getViewport().getSize().height;
+            duration = this.getTransitionDuration_(Math.abs(height),
+                viewportHeight);
+          },
+          () => {
+            st.setStyles(sectionChild, {
+              'opacity': '',
+              'height': 0,
+            });
+          });
+    }).then(() => {
+      return Animation.animate(this.element, tr.setStyles(sectionChild, {
+        'height': tr.px(tr.numeric(0, height)),
+      }), duration)
+          .thenAlways(() => {
+            this.mutateElement(() => {
+              st.setStyles(sectionChild, {
+                height: '',
+              });
+            });
+          });
+    });
+  }
+
+  /**
+   * @param {!Element} section
+   * @return {!Promise}
+   * @private
+   */
+  animateCollapse_(section) {
+    let height, duration;
+    const sectionChild = section.children[1];
```

Comment:
Does this intentionally grab the second child of `section`? Could be worth a comment explaining why it's `1` vs `0`

Reply:
Done